autoscaler/nanny/scaler.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nanny
+
+import (
+	"log"
+	"sort"
+	"strconv"
+)
+
+// Scaler determines the number of replicas to run
+type Scaler struct {
+	version   string
+	Verbose   bool
+	ladder    []int
+	scalerMap map[int]int32
+}
+
+func (s *Scaler) buildScalerMap(params *scalerParams) {
+	s.scalerMap = make(map[int]int32)
+	for k := range params.CoresToReplicasMap {
+		k64, err := strconv.ParseInt(k, 10, 32)
```

Comment:
ugh, yeah.  Yech.  OK.  Just pick something.  Suddenly JSON is not such a great answer :)